src/Microsoft.Identity.Client/Internal/MsalHelpers.cs

```diff
@@ -58,28 +58,14 @@ internal static string[] AsArray(this SortedSet<string> setOfStrings)
             return setOfStrings?.ToArray();
         }
 
-        internal static string AsSingleString(this SortedSet<string> setOfStrings)
+        internal static string AsSingleString(this IEnumerable<string> input)
         {
-            return AsSingleString(setOfStrings.ToArray());
-        }
-
-        internal static string AsSingleString(this string[] arrayOfStrings)
-        {
-            if (IsNullOrEmpty(arrayOfStrings))
+            if (IsNullOrEmpty(input))
             {
                 return String.Empty;
             }
 
-            StringBuilder sb = new StringBuilder();
-            sb.Append(arrayOfStrings[0]);
-
-            for (int i = 1; i < arrayOfStrings.Length; i++)
-            {
-                sb.Append(" ");
-                sb.Append(arrayOfStrings[i]);
-            }
-
-            return sb.ToString();
+            return string.Join(" ", input);
```

Comment:
_string.Join_ does the job of concatenating enumerables for us, and it's a fairly safe assumption such a common low-level function is bug-free and efficient (although you can check out the source code at https://referencesource.microsoft.com/#mscorlib/system/string.cs,fa961ce440526a4d).